client/client.js

```diff
@@ -62,71 +65,119 @@ Client.create = function (message) {
 // Instance methods
 
 // Persist subscriptions and presences when not already persisted in memory
-Client.prototype.storeData = function (message) {
-  var subscriptions = this.subscriptions,
-      presences = this.presences,
-      changed = false,
-      lookupMessage;
+Client.prototype.storeData = function (messageIn) {
+  var subCount = Object.keys(this.subscriptions).length,
+      presCount = Object.keys(this.presences).length,
+      processedOp = false;
+
+  // Persist the message data, according to type
+  switch(messageIn.op) {
+    case 'unsubscribe':
+    case 'sync':
+    case 'subscribe':
+      processedOp = this._storeDataSubscriptions(messageIn);
+      break;
+
+    case 'set':
+      processedOp = this._storeDataPresences(messageIn);
+      break;
+  }
+
+  // FIXME: For now log everything Later, enable sample logging. 
+  // if (processedOp && subCount % LOG_WINDOW_SIZE === 0) {
+  if (processedOp) {
+    log.info('#storeData', { 
+      client_id: this.id,
```

Comment:
minor nit, but this allocates a new object whether or not the log is consumed (whether through loglevel filtering, sampling, etc). Since this method is being called per-message, I consider it hot-path. Unless there's a benefit to logging objects, it would be lower overhead (fewer allocations, less GC churn) to log references to existing objects.